pkg/ipamd/ipamd.go

```diff
@@ -1321,3 +1344,21 @@ func (c *IPAMContext) SetNodeLabel(key, value string) error {
 func (c *IPAMContext) GetPod(podName, namespace string) (*v1.Pod, error) {
 	return c.k8sClient.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
 }
+
+// filterCNIUnmanagedENIs filters out ENIs on non-zero network cards such as p4 family
+func (c *IPAMContext) filterCNIUnmanagedENIs(enis []awsutils.ENIMetadata) []awsutils.ENIMetadata {
+	numFiltered := 0
+	ret := make([]awsutils.ENIMetadata, 0, len(enis))
+	for _, eni := range enis {
+		// If we have CNI unmanaged ENIs, filter them out
+		if c.awsClient.IsCNIUnmanagedENI(eni.ENIID) {
+			log.Debugf("Skipping ENI %s: since on non-zero network card", eni.ENIID)
+			numFiltered++
+			continue
+		}
+		ret = append(ret, eni)
+	}
+	c.cniunmanagedENI = numFiltered
+	c.updateIPStats(numFiltered)
```

Comment:
`updateIPStats ` updates the `enisMax ` and `ipMax` based on the absolute max minus what is passed as in input to this function.

```
func (c *IPAMContext) updateIPStats(unmanaged int) {
	ipMax.Set(float64(c.maxIPsPerENI * (c.maxENI - unmanaged)))
	enisMax.Set(float64(c.maxENI - unmanaged))
}
```

So this code conflicts and overrides what is done in function `filterUnmanagedENIs`. Whoever runs last (`filterCNIUnmanagedENIs` or `filterUnmanagedENIs`) will override the unmanaged number. This will be result in inaccurate `ipMax` and `enisMax` for node with multiple cards.

You can either merge the unmanaged ENIs in the same function (better approach) or update `updateIPStats` to handle this scenario.

Reply:
Yeah, I missed this part. Sure I can merge in the same function.

Reply:
Updated the PR with this.